FiniteStateMachine/src/demo/Car.java

```diff
@@ -35,12 +35,8 @@ public void reverse() {
         if (gearbox.getState() == Gear.REVERSE) {
             return;
         }
-        if (gearbox.nextStates().contains(Gear.REVERSE)) {
```

Comment:
fair, but the `Car` class was meant to illustrate some functions that can be written around a `StateMachine`. This example deliberately avoids calling a bad state transition. The fact that it just throws a `RuntimeException` instead is specific to this example method.